SU2_CFD/src/numerics/NEMO/convection/ausm_slau.cpp

```diff
@@ -0,0 +1,638 @@
+                                       CUpwAUSM_SLAU_Base_NEMO(val_nDim, val_nVar, val_nPrimVar,
+                                                               val_nPrimVarGrad, config) {
+
+  Minf = config->GetMach();
+  beta = 1.0/8.0;
+
+  if (Minf < EPS) SU2_MPI::Error("AUSM+M requires a reference Mach number (\"MACH_NUMBER\") greater than 0.", CURRENT_FUNCTION);
+
+}
+
+void CUpwAUSMPLUSM_NEMO::ComputeInterfaceQuantities(const CConfig* config, su2double *pressure,
+                                                    su2double &interface_mach,
+                                                    su2double *interface_soundspeed) {
+
+ 
+  const su2double sq_veli = GeometryToolbox::SquaredNorm(nDim, Velocity_i);  
+  const su2double sq_velj = GeometryToolbox::SquaredNorm(nDim, Velocity_j);
+
+  /*--- Calculate interface numerical gammas and speed of sound ---*/
+  const su2double Enthalpy_norm = 0.5*(Enthalpy_i+Enthalpy_j);
+  const su2double Gamma_ij = 0.5*(Gamma_i+Gamma_j);
+  const su2double A_s = sqrt(2.0*Enthalpy_norm*(Gamma_ij-1.0)/(Gamma_ij+1.0));
+
+  su2double A_F;
+  if (0.5*(ProjVelocity_i+ProjVelocity_j) >= 0.0) A_F = A_s*A_s/max(fabs(ProjVelocity_i), A_s);
+  else                                            A_F = A_s*A_s/max(fabs(ProjVelocity_j), A_s);
+
+  /*--- Compute L/R Mach numbers ---*/
+  interface_soundspeed[0] = interface_soundspeed[1] = A_F;
+  M_L = ProjVelocity_i/interface_soundspeed[0];
+  M_R = ProjVelocity_j/interface_soundspeed[1];
+
+  /*--- Interface mach number w/ pressure diffusion term (M_p)---*/
+  const su2double Density_F = 0.5*(Density_i+Density_j);
+  const su2double MF_sq = 0.5*(sq_veli+sq_velj)/(A_F*A_F);
+
+  const su2double param1 = max(MF_sq, Minf*Minf);
+  const su2double Mrefsq = (min(1.0, param1));
+  const su2double fa = 2.0*sqrt(Mrefsq)-Mrefsq;
+
+  const su2double alpha = 3.0/16.0*(-4.0+5.0*fa*fa);
+  const su2double f = 0.5*(1-cos(PI_NUMBER*min(1.0, max(abs(M_L), abs(M_R)))));
+
+  /*--- Pressure sensor terms ---*/
+  const su2double h = min(Sensor_i, Sensor_j);
+  const su2double g = 0.5*(1+cos(PI_NUMBER*h));
+  const su2double f0 = min(1.0, max(f, Minf*Minf));
+
+  /*--- Pressure diffusion term ---*/
+  const su2double M_p = -0.5*(1.0-f)*(Pressure_j-Pressure_i)/(Density_F*A_F*A_F)*(1.0-g);
+
+  /*--- Compute base split mach and pressure fluxes ---*/
+  su2double M_LP, M_RM, P_LP, P_RM;
+
+  if (fabs(M_L) <= 1.0) {
+    M_LP = 0.25*(M_L+1.0)*(M_L+1.0)+beta*(M_L*M_L-1.0)*(M_L*M_L-1.0);
+    P_LP = (0.25*(M_L+1.0)*(M_L+1.0)*(2.0-M_L)+alpha*M_L*(M_L*M_L-1.0)*(M_L*M_L-1.0));
+  } else { 
+    M_LP = 0.5*(M_L+fabs(M_L));
+    P_LP = 0.5*(M_L+fabs(M_L))/M_L;
+  }
+  
+  if (fabs(M_R) <= 1.0) {
+    M_RM = -0.25*(M_R-1.0)*(M_R-1.0)-beta*(M_R*M_R-1.0)*(M_R*M_R-1.0);
+    P_RM = (0.25*(M_R-1.0)*(M_R-1.0)*(2.0+M_R)-alpha*M_R*(M_R*M_R-1.0)*(M_R*M_R-1.0));
+  } else {
+    M_RM = 0.5*(M_R-fabs(M_R));
+    P_RM = 0.5*(M_R-fabs(M_R))/M_R;
+  }
```

Comment:
Some low-hanging cleanup by moving these to free functions (eventually can be shared with the ideal gas versions)

Reply:
Which file should these free functions live?

Reply:
You can keep them here (this file) for now I think

Reply:
So thinking about this more, these are not consistent between schemes.  Would the only purpose be too share between different solvers for the AUSM and AUSM+-Up2 scheme?

Reply:
correct, only where it is the same